Deployment types guidance for writers (#6745)
elastic/docs-content: contribute-docs/how-to/deployment-types/strategies.md

| Activity is atomic and required or performed in many other docs | Create a dedicated doc that covers the activity for all deployment types. Refer to that doc from other docs that need the activity. | • [](/deploy-manage/stack-settings.md)<br>• [](/deploy-manage/security/secure-settings.md) |
| Longer process that differs more by deployment type | Evaluate the trade-off between fitting everything into a single document and creating one document for each deployment type. Tools like [`applies-switch`](https://elastic.github.io/docs-builder/syntax/applies-switch/) tabs can help keep variation in one doc. See [Patterns by how a capability differs across deployment types](#patterns-by-how-a-capability-differs-across-deployment-types) for guidance. | • [](/deploy-manage/remote-clusters.md)<br>• [](/deploy-manage/tools/snapshot-and-restore.md) |
| Guide exists but only for some deployment types | Evaluate whether the scope is intentional (meets a business need) or a gap. If a gap, expand the doc to cover other deployment types. If it must stay scoped, tell the user why and offer alternatives or escape hatches. | [](/manage-data/migrate.md), currently scoped to ECH and ECE |

> | Pathway or process only exists for one or some deployment types | Break the content out into a labelable chunk: a bullet, a callout, or a section. Indicate applicability with [`applies_to`](/contribute-docs/how-to/cumulative-docs/index.md) tags. | [`elasticsearch-croneval`](elasticsearch://reference/elasticsearch/command-line-tools/elasticsearch-croneval.md), an {{es}} CLI utility for cron expression validation, only available on self-managed |

Reviewer: ```suggestion
| Pathway or process only exists for one or some deployment types | Break the content into elements you can label: bullets, callouts, or sections. Indicate applicability with [`applies_to`](/contribute-docs/how-to/cumulative-docs/index.md) tags. | [`elasticsearch-croneval`](elasticsearch://reference/elasticsearch/command-line-tools/elasticsearch-croneval.md), an {{es}} CLI utility for cron expression validation, only available on self-managed |
```

| **Replaced by a different mechanism.** Same user goal, different mechanism (most often {{serverless-short}}). | Shared overview of the user goal, then forked content for each mechanism. | Page-level `applies_to` on the overview. [`applies-switch`](https://elastic.github.io/docs-builder/syntax/applies-switch/) tabs if mechanisms are parallel; sibling pages if they're too different to coexist. |
| **Removed or not available.** Capability missing on some deployment types. | Don't just omit. Signal the absence where users are likely to look. | Don't add the missing type to page-level `applies_to`. Use section-level or inline `serverless: unavailable` where confusion is likely. Use the `removed` lifecycle value for version-specific removals. |

## Examples

> The following are real-world examples of how we chose to address varying processes across deployment types.

Reviewer: ```suggestion
The following are real-world examples of how we address varying processes across deployment types.
```

- [ ] **Prerequisites don't match the page scope.** A page tagged for one set of deployment types or versions has prerequisites tagged for a different set (not the same as [dimensions](/contribute-docs/how-to/cumulative-docs/guidelines.md#dimensions)).
- [ ] **Readers from other deployment types are stranded.** A page scoped to one or more deployment types (but not all) offers no cross-reference or explanation for readers using other deployment types. Link to the equivalent procedure or note why one doesn't exist.
- [ ] **A duplicated procedure where a cross-reference would do.** An atomic procedure (configure a setting, access {{kib}}, install an integration) is restated in multiple guides instead of being maintained in one place and linked to, or a procedure documented elsewhere is restated on a page without a snippet.
- [ ] **A missing feature on some deployment types is silently omitted.** A page covers a feature available on several deployment types but doesn't acknowledge that it's removed or replaced on others (most often {{serverless-short}}: for example, {{ilm-init}}, {{watcher}}, custom plugins, audit logging). Note the absence and link to the closest alternative.

> - [ ] **"Self-managed" used as a grouping label.** Some people refer to everything not hosted on {{ecloud}} as "self-managed", grouping ECK, ECE, and self-managed together. Don't use this grouping in our docs. ECK, ECE, and self-managed differ in meaningful ways that the grouping obscures. Always name the specific deployment type.

Reviewer 2: Use one of the suggestions [mentioned here](https://github.com/elastic/docs-content/pull/6745/changes#r3348966551) for consistency

> - [ ] **"Self-managed" used as a grouping label.** Some people refer to everything not hosted on {{ecloud}} as "self-managed", grouping ECK, ECE, and self-managed together. Don't use this grouping in our docs. ECK, ECE, and self-managed differ in meaningful ways that the grouping obscures. Always name the specific deployment type.

Author: ```suggestion
- [ ] **"Self-managed" used as a grouping label.** The term "self-managed" is sometimes used as a catch-all for anything not hosted on {{ecloud}}, including ECK and ECE. ECK, ECE, and self-managed are distinct deployment types with meaningful differences, and tying them together obscures those distinctions. Always refer to each deployment type by name.
```

:::{admonition} It's a marathon, not a sprint

> Sometimes, we don't have time to rework a tutorial to make it deployment-agnostic. Acknowledging the limitation and providing a way forward is a good first step. Open an issue or PR to track a longer-term fix.

Reviewer: ```suggestion
:::{admonition} Use an incremental, iterative approach
Sometimes, there's not enough time to rework a tutorial to make it deployment-agnostic. Acknowledging the limitation and providing a way forward is a good first step. Open an issue or PR to track a longer-term fix.
```

Reviewer: These are probably both matters of preference

> Sometimes, there's not enough time to rework a tutorial to make it deployment-agnostic. Acknowledging the limitation and providing a way forward is a good first step. Open an issue or PR to track a longer-term fix.

Author: ```suggestion
Sometimes, there's not enough time to rework a tutorial to make it deployment-agnostic. Acknowledging the limitation and providing a way forward is a good first step. Open an issue to track a longer-term fix.
```
